pkgs/development/python-modules/dkimpy/default.nix

```diff
@@ -1,38 +1,28 @@
 { stdenv, fetchurl, openssl, makeWrapper, buildPythonApplication
-, pytest, dnspython }:
+, pytest, dnspython, pynacl, authres }:
 
 let
-  majorversion = "0.6";
-  minorversion = "2";
+  majorversion = "0.8";
+  minorversion = "1";
 in buildPythonApplication rec {
   pname = "dkimpy";
   version = "${majorversion}.${minorversion}";
 
   src = fetchurl {
     url = "https://launchpad.net/${pname}/${majorversion}/${majorversion}.${minorversion}/+download/${pname}-${version}.tar.gz";
-    sha256 = "1hagz8qk0v4ijfbcdq4z28bpgr2mkpr498z76i1vam2d50chmakl";
+    sha256 = "125xakqg2j5jp4k5kafpnpazh9d97ysgayrsgpvm3nkdj4g8hw1j";
   };
 
   checkInputs = [ pytest ];
-  propagatedBuildInputs =  [ openssl dnspython ];
+  propagatedBuildInputs =  [ openssl dnspython pynacl authres ];
```

Comment:
`openssl` is not needed here

Reply:
My assumption was that since one of the tools (dknewkey) needs an openssl binary, this would have to be part of the path - and thus be a propagated input. Is that not the case?

Reply:
If it's needed in the path, then one of the following solutions should be used:
- Patch the path into the source code as you're doing it for `dknewkey` (preferred)
- Use `makeWrapper`
So, if the `substituteInPlace` you're doing patches all places where `openssl` is called you should be fine and won't need `openssl` in `*buildInputs`.

Reply:
But isn't it more robust to make it explicit that it needs this input? I can imagine some kind of future optimisation that creates minimal chroot environments with only the declared propagatedBuildInputs... There seems no harm in not having the very smallest closure if it is clearer for the future?

Reply:
Yes, there is harm. You don't want any project using dkimpy to openssl in its path. That should be made explicit.

Reply:
`propagatedBuildInputs` should generally never be used. Sadly, for Python libraries there's currently no better solution.